src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMStep.java

```diff
@@ -20,36 +15,29 @@
 import hudson.scm.SCM;
 import hudson.util.FormValidation;
 import hudson.util.ListBoxModel;
-import org.apache.commons.lang3.StringUtils;
 import org.jenkinsci.Symbol;
 import org.jenkinsci.plugins.workflow.steps.scm.SCMStep;
 import org.kohsuke.stapler.*;
 import org.kohsuke.stapler.verb.POST;
 
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import javax.inject.Inject;
 import java.util.*;
 
-import static java.lang.String.format;
 import static java.util.Collections.emptyList;
-import static java.util.Collections.singletonList;
 import static java.util.Objects.requireNonNull;
 import static org.apache.commons.lang3.StringUtils.isBlank;
 import static org.apache.commons.lang3.StringUtils.stripToNull;
 
 public class BitbucketSCMStep extends SCMStep {
 
     private List<BranchSpec> branches;
-    private String cloneUrl;
```

Comment:
These five fields (correctly) trip the UpgradeTest. All five are unsued- they were merely transitional variables used in creating the SCM itself. SCMSteps _can_ be persisted- however given the fields are kept as null until the constructor is called, and there's no setters for them, I don't _think_ this will invalidate any existing SCM Steps either persisted or declared in a pipeline script.
@dkjellin can you confirm? If so I'll mark this test as upgrade safe.